src/api/models/profiles/createProfile.ts

```diff
@@ -0,0 +1,54 @@
+import { Response } from 'express';
+import mongoose from 'mongoose';
+import { IProfileDoc, IProfile } from '../../../mongodb/profiles.db';
+import { patrolsRouter } from '../../routes/patrols.route';
+
+const Profile = mongoose.model<IProfileDoc>('Profiles');
+/**
+ * Creates a new profile with important attributes
+ * @param data The complete or incomplete data of the profile one would like to create.
+ * @param res A holder for any sent prescribed response from the server.
+ * @returns The id of the created profile.
+ */
+async function createProfile(data: IProfile, res: Response = null): Promise<IProfile> {
+  try {
+    //create a new profile document
+    const profile = new Profile({
+      name: data.name,
+      position: data.position,
+      patrol: data.patrol,
+      membership: {
+        troop: data.membership?.troop,
+        crew: data.membership?.crew,
+      },
+      health_form: data.health_form,
+      timestamps: {
+        created_at: data.timestamps?.created_at,
+        born_at: data.timestamps?.born_at,
+        joined_troop_at: data.timestamps?.joined_troop_at,
+        joined_crew_at: data.timestamps?.joined_crew_at,
+      },
+      hidden: data.hidden,
+    });
+    // saving the profile to the collection
+    await profile.save();
+
+    // let the server know the profile is saved
```

Comment:
this lets the _client_ know that the profile is saved

Reply:
this should be updated in all of the comments so it does not confuse anyone else that might look at our code